core/src/main/java/org/elasticsearch/search/internal/SearchContext.java

```diff
@@ -102,6 +103,7 @@ public static SearchContext current() {
     private Map<Lifetime, List<Releasable>> clearables = null;
     private final AtomicBoolean closed = new AtomicBoolean(false);
     private InnerHitsContext innerHitsContext;
+    private SearchTask task;
```

Comment:
It is probably worth some javadoc explaining why this isn't `final` and why it is ok that it isn't `volatile`.